website/server/models/challenge.js

```diff
@@ -149,7 +154,7 @@ schema.methods.syncToUser = async function syncChallengeToUser (user) {
     }
   });
 
-  toSave.push(user.save());
+  toSave.push(User.collection.update({_id: user._id}, user.toObject()));
   return Bluebird.all(toSave);
```

Comment:
We do have problems periodically with multiple updates that are supposed to happen at the same time but only some of them are executed or saved, leaving the user's data in an inconsistent state. From that point of view, it would be better to have one update instead of three.